src/Common/src/CoreLib/System/Resources/ResourceReader.cs

```diff
@@ -1124,6 +1139,57 @@ private string TypeNameFromTypeCode(ResourceTypeCode typeCode)
             }
         }
 
+        private static bool CompareNames(string typeName1, string typeName2)
+        {
+            // First, compare type names
+            int comma1 = typeName1.IndexOf(',');
+            int comma2 = typeName2.IndexOf(',');
+            if (comma1 != comma2)
+                return false;
+
+            // case sensitive
+            if (String.Compare(typeName1, 0, typeName2, 0, typeName2.Length, StringComparison.Ordinal) != 0)
+                return false;
+            if (comma1 == -1)
+                return true;
+
+            // Now, compare assembly display names (IGNORES VERSION AND PROCESSORARCHITECTURE)
+            // also, for  mscorlib ignores everything, since that's what the binder is going to do
+            while (Char.IsWhiteSpace(typeName1[++comma1]));
+            while (Char.IsWhiteSpace(typeName2[++comma2]));
+
+            // case insensitive
+            AssemblyName an1 = new AssemblyName(typeName1.Substring(comma1));
+            AssemblyName an2 = new AssemblyName(typeName1.Substring(comma2));
+            if (String.Compare(an1.Name, an2.Name, StringComparison.OrdinalIgnoreCase) != 0)
+                return false;
+
+            // to match IsMscorlib() in VM
+            if (String.Compare(an1.Name, "mscorlib", StringComparison.OrdinalIgnoreCase) == 0)
+                return true;
+
+
+            if ((an1.CultureInfo != null) && (an2.CultureInfo != null) &&
+                (an1.CultureInfo.Name != an2.CultureInfo.Name))
+                return false;
+
+            byte[] pkt1 = an1.GetPublicKeyToken();
+            byte[] pkt2 = an2.GetPublicKeyToken();
+            if ((pkt1 != null) && (pkt2 != null))
```

Comment:
I assume we want them both to be null / empty or both to have the same value? This can be done as a one-shot:

```cs
// don't need to null-check pkt1 or pkt2
if (!pkt1.AsSpan().SequenceEqual(pkt2.AsSpan())) { return false; }
```

Reply:
Sure, I can fix this.  FWIW this was code I copied from desktop.